src/Microsoft.DotNet.XHarness.CLI/Commands/Android/AndroidGetStateCommand.cs

```diff
@@ -25,24 +28,55 @@ public AndroidGetStateCommand() : base(TargetPlatform.Android, new ServiceCollec
 
     protected override Task<ExitCode> InvokeInternal(ILogger logger)
     {
-        var runner = new AdbRunner(logger);
-
-        logger.LogInformation("Getting state of ADB and attached Android device(s)");
         try
         {
-            string state = runner.GetAdbState();
-            if (string.IsNullOrEmpty(state))
+            var data = GetStateData(logger);
+
+            if (Arguments.UseJson)
             {
-                state = "No device attached";
-            }
-            logger.LogInformation($"ADB Version info:{Environment.NewLine}{runner.GetAdbVersion()}");
-            logger.LogInformation($"ADB State ('device' if physically attached):{Environment.NewLine}{state}");
+                var options = new JsonSerializerOptions
+                {
+                    WriteIndented = true
+                };
 
-            logger.LogInformation($"List of devices:");
-            var deviceAndArchList = runner.GetAttachedDevicesWithProperties("architecture");
-            foreach (string device in deviceAndArchList.Keys)
+                JsonSerializer.Serialize(Console.OpenStandardOutput(), data, options);
+            }
+            else
             {
-                logger.LogInformation($"Device: '{device}' - Architecture: {deviceAndArchList[device]}");
+                var state = data.DeviceState switch
+                {
+                    "device" => "Device/emulator is ready",
+                    null or "" => "No device attached",
+                    _ => data.DeviceState,
+                };
+
+                void PrintDeviceInfo(DeviceInfo device)

```

Comment:
A nit (even nittier when it's a local function) - a function that only logs would be better called "Log<something>" to make clear it just logs and does nothing else.

Reply:
Is not `Print` better?

For me `Log` means just dump it in the log file/stream/whatever. `Print` always had the connotation of "prettify for reading in console/on the screen" (which is what is happening here).

But maybe that's just my explanation?

Reply:
For me, I see jus call into logger.LogInformation() so for me it's technically logging, but as said, no big deal.